tutorials/Creating a new Lumigator endpoint.md

```diff
@@ -0,0 +1,581 @@
+might or might not become part of the public documentation - I think it might be useful for users to expand the tool
+with their own use cases and would allow us to grow more easily, but I am happy to collect feedback about it here first.
+
+## Lumigator's architecture
+
+The diagram below shows the current architecture of Lumigator. The larger
+containers (UI, Backend, Ray cluster, DB, Storage) are the different services Lumigator relies on.
+The components inside Backend, instead, are the different abstraction layers the backend itself relies on:
+
+* The **API** makes backend functionalities available to the UI (or whatever front-end you want to develop) through
+  different **routes** (code
+  [here](https://github.com/mozilla-ai/lumigator/tree/main/lumigator/python/mzai/backend/api/routes)).
+  [**Schemas**](https://github.com/mozilla-ai/lumigator/tree/main/lumigator/python/mzai/schemas) are used in the API
+  which allows one to exactly know which kind of data has to be passed to it (similarly, using the same schemas will be
+  a requirement for our SDK).
+
+* **Services** implement the actual functionalities and are called by the different methods exposed in the API (code
+  [here](https://github.com/mozilla-ai/lumigator/tree/main/lumigator/python/mzai/backend/services)).
+
+* **Repositories** implement the [repository pattern](https://www.cosmicpython.com/book/chapter_02_repository.html) as
+  an abstraction over the DB (code
+  [here](https://github.com/mozilla-ai/lumigator/tree/main/lumigator/python/mzai/backend/repositories)). They make use
+  of [record classes](https://github.com/mozilla-ai/lumigator/tree/main/lumigator/python/mzai/backend/records) to refer
+  to actual records on the DB.
+
+<p style="text-align: center;">
+<img src="assets/Lumigator%20Architecture%20-%2020240902.jpg" alt="Lumigator's architecture" >
+Lumigator's architecture
+</p>
+
+## Notation
+
+In the following, we will refer to paths inside Lumigator's repo relative to the `/lumigator/python/mzai` folder, e.g.
+the relative path `backend/api/routes/` (note the lack of a trailing slash) will map to the absolute path from the root
+of the repo `/lumigator/python/mzai/backend/api/routes/`.
+
+## Understanding Lumigator endpoints
+
+All the endpoints you can access in Lumigator's API are defined in
+[`backend/api/routes/`](https://github.com/mozilla-ai/lumigator/tree/main/lumigator/python/mzai/backend/api/routes)
+and explicitly listed in
+[`backend/api/router.py`](https://github.com/mozilla-ai/lumigator/blob/main/lumigator/python/mzai/backend/api/router.py),
+together with a metadata tag (defined
+[here](https://github.com/mozilla-ai/lumigator/blob/main/lumigator/python/mzai/backend/api/tags.py)) which is used to
+provide a short description of the route:
+
+<p style="text-align: center;">
+<img src="assets/image-20240909-091307.png" alt="Definition of the current routes in Lumigator" >
```

Comment:
suggestion: let's convene on a naming convention for assets. right now we have descriptive names, but could also use something like numbers, etc https://github.com/mozilla-ai/lumigator/tree/main/notebooks/assets

Reply:
Ah yes, in my case this was inherited from exporting a confluence page where I had pasted screenshots / uploaded images. Happy to rename them to descriptive names!